nginx-controller/nginx/nevermerger.go

```diff
@@ -0,0 +1,43 @@
+package nginx
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+type NeverMerger struct {
```

Comment:
We shouldn't have the NeverMerger, because it doesn't do merging, but keeps track of hostnames and Ingress resources. This doesn't fit into merging. I suggest to add its functionality to the Configurator.